### PR TITLE
Animator stopped notifying delegate to render when pipeline is not empty

### PR DESCRIPTION
**Pull request by ColdPaleLight**
fix https://github.com/flutter/flutter/issues/98959

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt. See [testing the engine] for instructions on
writing and running engine tests.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [x] All existing and new tests are passing.

### CODE REVIEW

File: shell/common/animator.cc [hunk 1]
```diff
@@ -175,9 +175,20 @@ void Animator::Render(std::unique_ptr<flutter::LayerTree> layer_tree) {
...
+  if (!result.success) {
     FML_DLOG(INFO) << "No pending continuation to commit";
+    return;
+  }
+
+  if (!result.is_first_item) {
+    // It has been successfully pushed to the pipeline but not as the first
+    // item. Eventually the 'Rasterizer' will consume it, so we don't need to
+    // notify the delegate.
+    FML_DLOG(INFO) << "Not the first item in the pipeline, ignore it";
```
Comment (dnfield) [inline shell/common/animator.cc:190]: nit: this log is likely to be noisy. I know it's only in local unopt builds, but I'm not sure what action I'd ever want to take if I was debugging a local engine build. We should drop it.

File: shell/common/pipeline.h [hunk 2]
```diff
@@ -179,32 +188,35 @@ class Pipeline {
...
     {
       std::scoped_lock lock(queue_mutex_);
+      is_first_item = queue_.empty();
       queue_.emplace_back(std::move(resource), trace_id);
     }
 
     // Ensure the queue mutex is not held as that would be a pessimization.
     available_.Signal();
-    return true;
+    return {/*success=*/true, /*is_first_item=*/is_first_item};
```
Comment (dnfield) [inline shell/common/pipeline.h:201]: nit: 
```cpp
return {.success=true, .is_first_item=is_first_item};
```

Comment (dnfield) [reply shell/common/pipeline.h:201]: (and below)

**Review by dnfield — COMMENTED** [review]
A few nits. @iskakaushik or @chinmaygarde may be interested in taking a look.

File: shell/common/animator.cc (hunk 1 above)
Comment (ColdPaleLight) [reply shell/common/animator.cc]: Done

File: shell/common/pipeline.h (hunk 2 above)
Comment (ColdPaleLight) [reply shell/common/pipeline.h]: Done

**Review by chinmaygarde — APPROVED** [review]

**Review by dnfield — APPROVED** [review]

**Pull request merged**